_posts/2019-02-02-digitalocean-one-click.md

```diff
@@ -0,0 +1,193 @@
+
+In this post we'll guide you through creating your first OpenFaaS function using the One-Click application.
+
+### Pre-Requisites
+
+* A DigitalOcean account
+
+An obvious pre-req is a [DigitalOcean](https://www.digitalocean.com/) account.  A DigitalOcean account with $100/60 days of credit can be  [created here](https://m.do.co/c/2962aa9e56a1).  This provides a generous amount of credit with the most economic Droplet size being just $5 a month.
+
+* The OpenFaaS CLI
+
+The OpenFaaS CLI (or `faas-cli`) is now a core piece of the OpenFaaS developer-experience and our users run it locally and in their CI jobs.  We'll use it to create and deploy our function onto the One-click Droplet.  The OpenFaaS docs provide all the detail you need to [get the CLI](https://docs.openfaas.com/cli/install/).
+
+### Create the Droplet
+
+[Log in](https://cloud.digitalocean.com/login) to your DigitalOcean account and head over to the create button, which details the products currently being offered by DigitalOcean.  The One-click apps reside under Droplets.
+
+![Create button menu](/images/digitalocean-one-click/create-button.png)
+
+On the resulting menu select the One-click apps option which will present a number of One-click apps.
+
+![Droplet menu](/images/digitalocean-one-click/droplet-menu.png)
+
+We'll choose the one labelled OpenFaaS on 18.04.  This will create a Swarm-based single node OpenFaaS instance on Ubuntu 18.04.
+
+![OpenFaaS One-click](/images/digitalocean-one-click/openfaas-oneclick.jpg)
+
+From here choose a Droplet size and region.  I'd recommend `4 GB / 2 CPUs / 80 GB SSD` and I tend to pick the region closest to me - being in the UK this is London.  Finally, choose your SSH key - if you need to generate and add a new key then please follow the [instructions on the previous blog](/blog/deploy-digitalocean-ansible/#create-and-upload-an-ssh-key).  With your chosen configuration click Create.
+
+![Click create](/images/digitalocean-one-click/create-droplet.png)
+
+### Access the Droplet
+
+Once the Droplet is ready it will appear in your list of Droplets within the DigitalOcean resources pane.
+
+![Droplet is ready](/images/digitalocean-one-click/droplet-ready.png)
+
+Copy the IP address and using your favourite terminal log in to the Droplet using `root@<IP address>`
+
+> Note: ensure you have made the appropriate private key available to the client you are using.
+
+Once logged in follow the instructions; at the time of writing this is simply to press Enter.
+
+![Logged in](/images/digitalocean-one-click/terminal.png)
+
+A number of configuration actions will be performed for you, and the result will be a running Swarm-based OpenFaaS instance.  Make a note of the credentials - in particular the final line which will set our credentials in the `faas-cli` - we'll need this when we deploy our function.  We can now close the connection to the Droplet.
+
+* On your local machine set the `OPENFAAS_URL`:
```

Comment:
We could mention they can edit their `.bashrc` file or `.bash_profile` to retain the setting / make permanent.